engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/task/UserTaskProcessor.java

```diff
@@ -0,0 +1,135 @@
+
+  public UserTaskProcessor(
+      final BpmnBehaviors bpmnBehaviors,
+      final BpmnStateTransitionBehavior stateTransitionBehavior) {
+    zeebeUserTaskBehavior = new ZeebeUserTaskBehavior(bpmnBehaviors, stateTransitionBehavior);
+    jobWorkerTaskBehavior = new JobWorkerTaskBehavior(bpmnBehaviors, stateTransitionBehavior);
+  }
+
+  @Override
+  public Class<ExecutableUserTask> getType() {
+    return ExecutableUserTask.class;
+  }
+
+  @Override
+  public void onActivate(final ExecutableUserTask element, final BpmnElementContext context) {
+    eventBehaviorOf(element, context).onActivate(element, context);
+  }
+
+  @Override
+  public void onComplete(final ExecutableUserTask element, final BpmnElementContext context) {
+    eventBehaviorOf(element, context).onComplete(element, context);
+  }
+
+  @Override
+  public void onTerminate(final ExecutableUserTask element, final BpmnElementContext context) {
+    eventBehaviorOf(element, context).onTerminate(element, context);
+  }
+
+  private UserTaskBehavior eventBehaviorOf(
+      final ExecutableUserTask element, final BpmnElementContext context) {
+    if (element.getJobWorkerProperties() != null) {
+      return jobWorkerTaskBehavior;
+    } else if (element.getUserTaskProperties() != null) {
+      return zeebeUserTaskBehavior;
+    } else {
+      throw new BpmnProcessingException(
+          context, "Expected to process user task, but could not determine processing behavior");
+    }
+  }
+
+  private static final class ZeebeUserTaskBehavior implements UserTaskBehavior {
+
+    private final BpmnIncidentBehavior incidentBehavior;
+    private final BpmnStateTransitionBehavior stateTransitionBehavior;
+    private final BpmnVariableMappingBehavior variableMappingBehavior;
+    private final BpmnEventSubscriptionBehavior eventSubscriptionBehavior;
+    private final BpmnStateBehavior stateBehavior;
+    private final ExpressionProcessor expressionProcessor;
+
+    private final EventTriggerBehavior eventTriggerBehavior;
+
+    public ZeebeUserTaskBehavior(
+        final BpmnBehaviors bpmnBehaviors,
+        final BpmnStateTransitionBehavior stateTransitionBehavior) {
+      eventSubscriptionBehavior = bpmnBehaviors.eventSubscriptionBehavior();
+      incidentBehavior = bpmnBehaviors.incidentBehavior();
+      this.stateTransitionBehavior = stateTransitionBehavior;
+      variableMappingBehavior = bpmnBehaviors.variableMappingBehavior();
+      stateBehavior = bpmnBehaviors.stateBehavior();
+      expressionProcessor = bpmnBehaviors.expressionBehavior();
+      eventTriggerBehavior = bpmnBehaviors.eventTriggerBehavior();
+    }
+
+    @Override
+    public void onActivate(final ExecutableUserTask element, final BpmnElementContext context) {}
+
+    @Override
+    public void onComplete(final ExecutableUserTask element, final BpmnElementContext context) {}
+
+    @Override
+    public void onTerminate(final ExecutableUserTask element, final BpmnElementContext context) {}
```

Comment:
:upside_down_face: Should we add TODO comments here that reference the GitHub issues that will implement them?

That way it would be clear why they remain unimplemented to other Zeebe devs.

Reply:
Oh yeah, good idea 👍

Reply:
Added todo comments 👍